great_ape_safe/great_ape_safe.py

```diff
@@ -51,21 +52,13 @@ def __init__(self, address, base_url=None, multisend=None):
 
 
     def init_all(self):
-        self.init_aave()
-        self.init_anyswap()
-        self.init_badger()
-        self.init_compound()
-        self.init_convex()
-        self.init_cow()
-        self.init_curve()
-        self.init_curve_v2()
-        self.init_opolis()
-        self.init_pancakeswap_v2()
-        self.init_rari()
-        self.init_solidly()
-        self.init_sushi()
-        self.init_uni_v2()
-        self.init_uni_v3()
+        for method in self.__dir__():
+            if method.startswith('init_') and method != 'init_all':
+                try:
+                    getattr(self, method)()
+                except exceptions.ContractNotFound:
+                    # different chain
+                    pass
```

Comment:
very nice!

there is actually another optimisation i have been thinking about for a while: all the constants in the protocol classes such as `self.max_slippage` and `self.deadline` should actually be declared outside the `__init__` block. this will make them inheritable, saving having to add them manually to every time ;)

maybe you can push that through in this pr as well, or write a new ticket for it?

Reply:
good idea, will add in this pr